internal/auth0/user/resource_permission.go

```diff
@@ -0,0 +1,169 @@
+package user
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/auth0/terraform-provider-auth0/internal/mutex"
+)
+
+// NewPermissionResource will return a new auth0_connection_client resource.
+func NewPermissionResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID of user to associate the permission to.",
+			},
+			"permission": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the permission.",
+			},
+			"resource_server_identifier": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Identifier (ID) of the resource server that the permission is associated with.",
```

Comment:
The identifier is not the resource server ID, they're 2 separate things. However, is the ID also accepted by the API?

![image](https://github.com/auth0/terraform-provider-auth0/assets/28300158/46df715a-fb60-4456-ba05-ab0554a29a01)